compiler/test/fail_compilation/fail12236.d

```diff
@@ -7,7 +7,7 @@ fail_compilation/fail12236.d(21): Error: forward reference to inferred return ty
 fail_compilation/fail12236.d(21):        while evaluating `pragma(msg, f2(T)(T).mangleof)`
 fail_compilation/fail12236.d(27): Error: template instance `fail12236.f2!int` error instantiating
 fail_compilation/fail12236.d(31): Error: forward reference to inferred return type of function `__lambda1`
-fail_compilation/fail12236.d(31):        while evaluating `pragma(msg, __lambda1.mangleof)`
+fail_compilation/fail12236.d(31):        while evaluating `pragma(msg, __lambda1(__T1)(a).mangleof)`
```

Comment:
If we use a template declaration instead of a lambda, the message here would contain the template parameters so this should be fine.